Code/Framework/AzToolsFramework/AzToolsFramework/UnitTest/Mocks/MockEditorViewportIconDisplayInterface.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzToolsFramework/API/EditorViewportIconDisplayInterface.h>
+
+#include <gmock/gmock.h>
+
+namespace UnitTest
+{
+    class MockEditorViewportIconDisplayInterface : public AzToolsFramework::EditorViewportIconDisplayInterface
+    {
+    public:
+        virtual ~MockEditorViewportIconDisplayInterface() = default;
+
+        void Register()
+        {
+            AZ::Interface<EditorViewportIconDisplayInterface>::Register(this);
+        }
+
+        void Deregister()
+        {
+            AZ::Interface<EditorViewportIconDisplayInterface>::Unregister(this);
+        }
```

Comment:
In case you find this useful, the class could inherit from `public AzToolsFramework::EditorViewportIconDisplayInterface::Registrar` and it would the the registration automatically for you on instantiation/destruction, functions `Register` and `Deregister` could be removed.

Reply:
Oh interesting! I was not aware of this, I can make this update as it avoids some additional boilerplate, thanks for the suggestion! 👍